src/util/carddb.ts

```diff
@@ -111,13 +111,21 @@ export function getMostReasonable(
   printing: PrintingPreference = PrintingPreference.RECENT,
   filter?: FilterFunction,
 ): CardDetails | null {
-  let ids = getIdsFromName(cardName);
+  const ids = getIdsFromName(cardName);
   if (ids === undefined || ids.length === 0) {
     // Try getting it by ID in case this is an ID.
 
     return getMostReasonableById(cardName, printing);
   }
 
+  return getMostReasonableByPrintingPreference(ids, printing, filter);
+}
+
+export function getMostReasonableByPrintingPreference(
```

Comment:
In my experience I've found optional parameters can get out of hand quickly, whereas alternate versions of the function with descriptive name differences is easier. Unless you get to having so many parameters that it makes sense to define a config object type just for it.

But if you'd prefer consolidating such that printing preference could be set or not, I can refactor.